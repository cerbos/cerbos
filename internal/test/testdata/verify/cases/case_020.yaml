description: Valid YAML test suite with multiple principals and resources in expectations
