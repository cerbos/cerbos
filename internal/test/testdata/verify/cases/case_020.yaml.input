-- testdata/principals.yaml --
---
principals:
  bev: &bev
    id: bev
    policyVersion: '20210210'
    roles:
      - employee
      - manager
    attr:
      department: marketing
      geography: GB
      managed_geographies: GB
      ip_address: 10.20.1.2
      team: design
  barb:
    <<: *bev
    id: barb
  bob:
    <<: *bev
    id: bob

-- testdata/resources.yaml --
---
resources:
  john_leave_request:
    kind: leave_request
    policyVersion: '20210210'
    id: XX125
    attr: &attr
      department: marketing
      geography: GB
      id: XX125
      owner: john
      team: design
  pending_leave_request:
    kind: leave_request
    policyVersion: '20210210'
    id: XX125
    attr:
      <<: *attr
      status: PENDING_APPROVAL
  stale_leave_request:
    kind: leave_request
    policyVersion: '20210210'
    id: XX225
    attr:
      <<: *attr
      modifiedAt: "2022-08-01T15:00:00Z"
  stale_pending_leave_request:
    kind: leave_request
    policyVersion: '20210210'
    id: XX225
    attr:
      <<: *attr
      modifiedAt: "2022-08-01T15:00:00Z"
      status: PENDING_APPROVAL

-- testdata/auxdata.yaml --
---
auxData:
  myJWT:
    jwt:
      iss: cerbos-test-suite
      aud: [cerbos-jwt-tests]
      customArray: [A, B]

-- suite_test.yaml --
---
name: TestSuite
description: Tests for verifying something
options:
  now: "2022-08-02T15:00:00Z"
tests:
  - name: Multiple principals
    input:
      principals:
        - bev
        - barb
      resources:
        - pending_leave_request
      actions:
        - approve
        - delete
      auxData: myJWT
    expected:
      - principals:
          - bev
          - barb
        resource: pending_leave_request
        actions:
          approve: EFFECT_ALLOW
          delete: EFFECT_ALLOW

  - name: Multiple resources
    input:
      principals:
        - bob
      resources:
        - pending_leave_request
        - stale_pending_leave_request
      actions:
        - approve
        - delete
      auxData: myJWT
    expected:
      - principal: bob
        resources:
          - pending_leave_request
          - stale_pending_leave_request
        actions:
          approve: EFFECT_ALLOW
          delete: EFFECT_ALLOW

  - name: Multiple principals and resources
    input:
      principals:
        - bev
        - barb
        - bob
      resources:
        - stale_leave_request
        - stale_pending_leave_request
      actions:
        - remind
        - view:public
      auxData: myJWT
    expected:
      - principals:
          - bev
          - barb
          - bob
        resources:
          - stale_leave_request
          - stale_pending_leave_request
        actions:
          remind: EFFECT_ALLOW
          "view:public": EFFECT_ALLOW
