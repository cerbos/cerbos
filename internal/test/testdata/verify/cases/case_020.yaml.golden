{
  "suites": [
    {
      "file": "suite_test.yaml",
      "name": "TestSuite",
      "summary": {
        "overallResult": "RESULT_PASSED",
        "testsCount": 20,
        "resultCounts": [
          {
            "result": "RESULT_PASSED",
            "count": 20
          }
        ]
      },
      "testCases": [
        {
          "name": "Multiple principals",
          "principals": [
            {
              "name": "bev",
              "resources": [
                {
                  "name": "pending_leave_request",
                  "actions": [
                    {
                      "name": "approve",
                      "details": {
                        "result": "RESULT_PASSED",
                        "success": {
                          "effect": "EFFECT_ALLOW"
                        }
                      }
                    },
                    {
                      "name": "delete",
                      "details": {
                        "result": "RESULT_PASSED",
                        "success": {
                          "effect": "EFFECT_ALLOW"
                        }
                      }
                    }
                  ]
                }
              ]
            },
            {
              "name": "barb",
              "resources": [
                {
                  "name": "pending_leave_request",
                  "actions": [
                    {
                      "name": "approve",
                      "details": {
                        "result": "RESULT_PASSED",
                        "success": {
                          "effect": "EFFECT_ALLOW"
                        }
                      }
                    },
                    {
                      "name": "delete",
                      "details": {
                        "result": "RESULT_PASSED",
                        "success": {
                          "effect": "EFFECT_ALLOW"
                        }
                      }
                    }
                  ]
                }
              ]
            }
          ]
        },
        {
          "name": "Multiple resources",
          "principals": [
            {
              "name": "bob",
              "resources": [
                {
                  "name": "pending_leave_request",
                  "actions": [
                    {
                      "name": "approve",
                      "details": {
                        "result": "RESULT_PASSED",
                        "success": {
                          "effect": "EFFECT_ALLOW"
                        }
                      }
                    },
                    {
                      "name": "delete",
                      "details": {
                        "result": "RESULT_PASSED",
                        "success": {
                          "effect": "EFFECT_ALLOW"
                        }
                      }
                    }
                  ]
                },
                {
                  "name": "stale_pending_leave_request",
                  "actions": [
                    {
                      "name": "approve",
                      "details": {
                        "result": "RESULT_PASSED",
                        "success": {
                          "effect": "EFFECT_ALLOW"
                        }
                      }
                    },
                    {
                      "name": "delete",
                      "details": {
                        "result": "RESULT_PASSED",
                        "success": {
                          "effect": "EFFECT_ALLOW"
                        }
                      }
                    }
                  ]
                }
              ]
            }
          ]
        },
        {
          "name": "Multiple principals and resources",
          "principals": [
            {
              "name": "bev",
              "resources": [
                {
                  "name": "stale_leave_request",
                  "actions": [
                    {
                      "name": "remind",
                      "details": {
                        "result": "RESULT_PASSED",
                        "success": {
                          "effect": "EFFECT_ALLOW"
                        }
                      }
                    },
                    {
                      "name": "view:public",
                      "details": {
                        "result": "RESULT_PASSED",
                        "success": {
                          "effect": "EFFECT_ALLOW"
                        }
                      }
                    }
                  ]
                },
                {
                  "name": "stale_pending_leave_request",
                  "actions": [
                    {
                      "name": "remind",
                      "details": {
                        "result": "RESULT_PASSED",
                        "success": {
                          "effect": "EFFECT_ALLOW"
                        }
                      }
                    },
                    {
                      "name": "view:public",
                      "details": {
                        "result": "RESULT_PASSED",
                        "success": {
                          "effect": "EFFECT_ALLOW"
                        }
                      }
                    }
                  ]
                }
              ]
            },
            {
              "name": "barb",
              "resources": [
                {
                  "name": "stale_leave_request",
                  "actions": [
                    {
                      "name": "remind",
                      "details": {
                        "result": "RESULT_PASSED",
                        "success": {
                          "effect": "EFFECT_ALLOW"
                        }
                      }
                    },
                    {
                      "name": "view:public",
                      "details": {
                        "result": "RESULT_PASSED",
                        "success": {
                          "effect": "EFFECT_ALLOW"
                        }
                      }
                    }
                  ]
                },
                {
                  "name": "stale_pending_leave_request",
                  "actions": [
                    {
                      "name": "remind",
                      "details": {
                        "result": "RESULT_PASSED",
                        "success": {
                          "effect": "EFFECT_ALLOW"
                        }
                      }
                    },
                    {
                      "name": "view:public",
                      "details": {
                        "result": "RESULT_PASSED",
                        "success": {
                          "effect": "EFFECT_ALLOW"
                        }
                      }
                    }
                  ]
                }
              ]
            },
            {
              "name": "bob",
              "resources": [
                {
                  "name": "stale_leave_request",
                  "actions": [
                    {
                      "name": "remind",
                      "details": {
                        "result": "RESULT_PASSED",
                        "success": {
                          "effect": "EFFECT_ALLOW"
                        }
                      }
                    },
                    {
                      "name": "view:public",
                      "details": {
                        "result": "RESULT_PASSED",
                        "success": {
                          "effect": "EFFECT_ALLOW"
                        }
                      }
                    }
                  ]
                },
                {
                  "name": "stale_pending_leave_request",
                  "actions": [
                    {
                      "name": "remind",
                      "details": {
                        "result": "RESULT_PASSED",
                        "success": {
                          "effect": "EFFECT_ALLOW"
                        }
                      }
                    },
                    {
                      "name": "view:public",
                      "details": {
                        "result": "RESULT_PASSED",
                        "success": {
                          "effect": "EFFECT_ALLOW"
                        }
                      }
                    }
                  ]
                }
              ]
            }
          ]
        }
      ],
      "description": "Tests for verifying something"
    }
  ],
  "summary": {
    "overallResult": "RESULT_PASSED",
    "testsCount": 20,
    "resultCounts": [
      {
        "result": "RESULT_PASSED",
        "count": 20
      }
    ]
  }
}
